File: packages/silo-client/src/collections/collection-definition.ts
import type { JsonSchema } from "./json-schema.js";

/** One collection with its schema, bundled so `silo://` refs are already
 * resolved. What `create`, `schema.get()`, `schema.put()` and
 * `environment.schemas()` all answer. No mapping needed: every key here is
 * already the wire's own spelling. */
export interface CollectionDefinition {
  readonly id: string;
  readonly name: string;
  readonly schema: JsonSchema;
}

File: packages/silo-client-dart/lib/src/collections/collection_definition.dart
import 'json_schema.dart';

/// A collection with its schema, `silo://` refs already resolved.
final class CollectionDefinition {
  const CollectionDefinition(this.id, this.name, this.schema);

  factory CollectionDefinition.fromJson(Map<String, Object?> json) =>
      CollectionDefinition(json['id'] as String, json['name'] as String, json['schema'] as JsonSchema);

  final String id;
  final String name;
  final JsonSchema schema;
}
